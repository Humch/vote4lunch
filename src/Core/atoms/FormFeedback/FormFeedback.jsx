import React from 'react';

import PropTypes from 'prop-types';

const FormFeedback = ({ valid, text }) => (
  <div className={valid ? 'valid-feedback' : 'invalid-feedback'}>
    {text}
  </div>
);

FormFeedback.propTypes = {
  valid: PropTypes.bool,
  text: PropTypes.string.isRequired,
};

export default FormFeedback;
