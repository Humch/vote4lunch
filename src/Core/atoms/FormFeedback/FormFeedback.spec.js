/* eslint-disable no-undef */
import React from 'react';

import renderer from 'react-test-renderer';

import FormFeedback from './FormFeedback';


describe('FormFeedback Snapshot', () => {
  it('capturing Snapshot of FormFeedback with valid equal false', () => {
    const renderedValue = renderer.create(
      <FormFeedback
        valid={false}
        text="Ce champ est invalide ;)"
      />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('capturing Snapshot of ValueChecker with valid', () => {
    const renderedValue = renderer.create(
      <FormFeedback
        valid
        text="Ce champ est valide ;)"
      />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
