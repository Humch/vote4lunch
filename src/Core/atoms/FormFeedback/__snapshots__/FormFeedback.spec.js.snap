// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFeedback Snapshot capturing Snapshot of FormFeedback with valid equal false 1`] = `
<div
  className="invalid-feedback"
>
  Ce champ est invalide ;)
</div>
`;

exports[`FormFeedback Snapshot capturing Snapshot of ValueChecker with valid 1`] = `
<div
  className="valid-feedback"
>
  Ce champ est valide ;)
</div>
`;
